pkg/services/authn/clients/jwt.go

```diff
@@ -78,10 +78,17 @@ func (s *JWT) Authenticate(ctx context.Context, r *authn.Request) (*authn.Identi
 	if key := s.cfg.JWTAuth.UsernameClaim; key != "" {
 		id.Login, _ = claims[key].(string)
 		id.ClientParams.LookUpParams.Login = &id.Login
+	} else if key := s.cfg.JWTAuth.UsernameAttributePath; key != "" {
+		id.Login, _ = util.SearchJSONForStringAttr(s.cfg.JWTAuth.UsernameAttributePath, claims)
```

Comment:
We need to handle potential errors for both of these

Reply:
I intentionally didn't handle the errors similar to what was already being done 3 lines before.
```
id.Login, _ = claims[key].(string)
```

They don't handle the errors because the identity validation expects username and email to be set.


If we handle errors for these 2, we might as well handle the errors for the other 2. WDYT @kalleep ?

Reply:
@linoman The emptiness of `id.Login` and `id.Email` is checked at the end of this function: 
https://github.com/grafana/grafana/blob/29c76b98d4901a7979edab3b9267c870109b6c34/pkg/services/authn/clients/jwt.go#L120-L123

However, I think we should still check/log and return the errors here (for example when the JMESPath is incorrect) as Kalle said.